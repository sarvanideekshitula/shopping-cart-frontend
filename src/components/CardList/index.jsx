import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './index.module.css';
import Card from '../Card/index';

const CardList = (props) => {
  const { products, testId } = props;
  const cardList = products.map((value) => <Card id={value.id} name={value.name} price={value.price} quantity={value.quantity} link={value.imageLink} selectedQuantity={value.selectedQuantity} />);
  return (
    <div className={styles.CardList} data-testId={testId}>
      {cardList}
    </div>
  );
};
CardList.propTypes = {
  products: PropTypes.objectOf.isRequired,
  testId: PropTypes.string.isRequired,
};
export default CardList;
