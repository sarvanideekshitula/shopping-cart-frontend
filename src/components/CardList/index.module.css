.CardList{
    display: flex;
    justify-content: space-around;
    flex-direction: row;
    flex-wrap: wrap;
    margin-left: 24rem;
    margin-right: 24rem;
    margin-top: 4rem;
    width: 50%;
}