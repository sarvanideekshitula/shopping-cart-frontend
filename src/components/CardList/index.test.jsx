import React from 'react';
import { render } from '@testing-library/react';
import CardList from './index';

describe('the Genre container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<CardList products={[]} testId="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
