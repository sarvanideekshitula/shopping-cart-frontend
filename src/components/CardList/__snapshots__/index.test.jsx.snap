// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Genre container should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="test"
  />
</DocumentFragment>
`;
